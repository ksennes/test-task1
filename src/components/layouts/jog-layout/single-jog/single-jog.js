import React from "react";
import {fromUnixTime, format} from 'date-fns';

import "./single-jog.scss";
import icon from "../../../../img/icon.png";

export const SingleJog = ({ date, distance, time }) => {
  return (
    <div className="single-jog">
      <div className="single-jog__icon">
        <img src={icon} />
      </div>
      <div className="single-jog__details">
        <p className="date">{date}</p>
        <h4>Distance: <span>{distance}</span></h4>
        <h4>Time: <span>{time}</span></h4>
      </div>
    </div>
  );
};
