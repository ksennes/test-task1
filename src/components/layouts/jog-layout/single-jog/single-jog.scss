.single-jog {
  display: flex;
  align-items: center;
  margin-top: 50px;
  font-size: 14px;

  &__icon {
    margin-right: 46px;
  }

  &__details {
    p {
      color: #747474;
      margin-bottom: 9px;
    }

    h4 {
      margin-bottom: 12px;
    }

    span {
      color: #747474;
      font-weight: normal;
    }
  }
}

@media (max-width: 375px) {
  .single-jog {
    margin: 45px 0;
    align-items: flex-start;
  }
}