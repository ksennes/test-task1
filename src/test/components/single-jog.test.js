import React from 'react';
import { SingleJog } from '../../components/layouts/jog-layout/single-jog/single-jog';
import renderer from 'react-test-renderer';

describe('SingleJog component', () => {
  it('sould match the last snapshot without jog', () => {
    const wrapper = renderer.create(<SingleJog />)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('sould match the last snapshot with jog', () => {
    const jog = {
     date: '20.02.2020', 
     distance: '20', 
     time: '20'
    }
    const wrapper = renderer.create(<SingleJog jog={jog}/>)
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});