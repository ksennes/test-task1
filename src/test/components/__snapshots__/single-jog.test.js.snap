// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleJog component sould match the last snapshot with jog 1`] = `
<div
  className="single-jog"
>
  <div
    className="single-jog__icon"
  >
    <img
      src="test-file-stub"
    />
  </div>
  <div
    className="single-jog__details"
  >
    <p
      className="date"
    />
    <h4>
      Distance: 
      <span />
    </h4>
    <h4>
      Time: 
      <span />
    </h4>
  </div>
</div>
`;

exports[`SingleJog component sould match the last snapshot without jog 1`] = `
<div
  className="single-jog"
>
  <div
    className="single-jog__icon"
  >
    <img
      src="test-file-stub"
    />
  </div>
  <div
    className="single-jog__details"
  >
    <p
      className="date"
    />
    <h4>
      Distance: 
      <span />
    </h4>
    <h4>
      Time: 
      <span />
    </h4>
  </div>
</div>
`;
